### PR TITLE
Include entity version data to the `QueryResponse`

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR addresses #683.

`QueryResponse` now stores `EntityStateWithVersion` instances instead of just entity state:
```proto
message QueryResponse {
    ...
    repeated EntityStateWithVersion messages = 5;
}

message EntityStateWithVersion {
    google.protobuf.Any state = 1;
    core.Version version = 2;
}
```

To enable this functionality, the `MirrorProjection` and related events now also store the corresponding aggregate version.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: client/src/main/proto/spine/client/query.proto [hunk 1]
```diff
@@ -87,9 +88,21 @@ message QueryResponse {
     // Reserved for more query response attributes, e.g. to describe paginated response etc.
     reserved 2 to 4;
 
-    //TODO:2017-02-20:alexander.yevsyukov: Use EntityRecord instead of Any.
```
Comment (armiol) [inline client/src/main/proto/spine/client/query.proto]: Please move the `todo` item into the `EntityStateWithVersion`.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL again. The `web` will need a couple of minor fixes, but is compatible with this change overall.

**Comment by armiol** [comment]
@dmitrykuzmin I have recalled why we had left this change out.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL again.

**Pull request merged**